protocols/mdns/src/behaviour.rs

```diff
@@ -304,6 +308,11 @@ where
         &mut self,
         cx: &mut Context<'_>,
     ) -> Poll<ToSwarm<Self::ToSwarm, THandlerInEvent<Self>>> {
+        // Checking for pending events and emit them
+        if let Some(event) = self.pending_events.pop_front() {
+            return Poll::Ready(event);
+        }
```

Comment:
For the sake of consistency, can we also push the `Event::Expired` (L.400) to this queue instead of returning it directly?